packages/flutter/lib/src/material/tabs.dart

```diff
@@ -557,8 +558,18 @@ class _TabBarState extends State<TabBar> {
     });
   }
 
+  // Called each time layout completes.
   void _saveTabOffsets(List<double> tabOffsets) {
     _indicatorPainter?.tabOffsets = tabOffsets;
+
+    if (!_completedInitialScroll) {
+      _completedInitialScroll = true;
+      if (widget.isScrollable && _currentIndex != 0) {
+        scheduleMicrotask(() {
```

Comment:
At this point, although layout of the TabBar's tabs is complete, the ScrollController's position.viewportDimension is still null.  So I've deferred doing the scroll until layout is finished, at which point the viewportDimension has been set.

If there's another way to get this to go, I'm all ears.